nova_plugin/server.py

```diff
@@ -721,6 +743,28 @@ def ud_http(params):
         ('url',),
         "server.userdata when using type 'http'")
     return requests.get(params['url']).text
+
+
+@userdata_handler('file')
```

Comment:
this code went through refactoring in 3.3 (1.3 plugin version), if we want to merge it there then the new code here will also have to get refactored